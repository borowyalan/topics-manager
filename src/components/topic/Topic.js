import React from 'react';
import TopicDescription from "./TopicDescription";
import TrainersList from "./TrainersList";
import styled from 'styled-components';
import Participants from "./Participants";

const Topic = props => (
    <RowWrapper>
        <TopicDescription/>
        <TrainersList/>
        <Participants/>
    </RowWrapper>
);

const RowWrapper = styled.li`
    width: 100%;
    margin-top: 2em;
    display: flex;
    flex: start;
    justify-content: space-between;
    text-align: left;
`;

export default Topic;
