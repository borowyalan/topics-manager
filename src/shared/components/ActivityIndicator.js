import React from 'react';

const ActivityIndicator = () => {
    return (
        <p>Loading</p>
    );
};

export default ActivityIndicator;
