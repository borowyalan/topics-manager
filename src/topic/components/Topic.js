import React from 'react';
import TopicDescription from './TopicDescription';
import TrainersList from './TrainersList';
import styled from 'styled-components';
import Participants from './Participants';
import PropTypes from 'prop-types';
import TopicModel from '../../shared/models/TopicModel';

const Topic = ({topic}) => (
    <RowWrapper>
        <TopicDescription title={topic.title} description={topic.description}/>
        <TrainersList trainers={topic.trainers} author={topic.author}/>
        <Participants topic={topic}/>
    </RowWrapper>
);

Topic.propTypes = {
    topic: PropTypes.instanceOf(TopicModel).isRequired
};

const RowWrapper = styled.p`
    width: 100%;
    margin-top: 2em;
    display: flex;
    flex: start;
    justify-content: space-between;
    text-align: left;
`;

export default Topic;
