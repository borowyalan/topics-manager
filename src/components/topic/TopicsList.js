import React from 'react';
import Section from '../Section';
import Text from '../Text';
import Topic from './Topic';

const TopicsList = props => (
    <Section>
        <Text type="primary">Zg≈Çoszone tematy</Text>
        <Topic/>
    </Section>
);

export default TopicsList;
