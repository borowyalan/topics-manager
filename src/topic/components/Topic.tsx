import React, { SFC } from 'react';
import styled from 'styled-components';
import { Topic } from '../../shared/models/topic';
import Participants from './Participants';
import TopicDescription from './TopicDescription';
import TrainersList from './TrainersList';

export interface TopicProps {
    topic: Topic
}

const Topic: SFC<TopicProps> = ({ topic }) => (
    <RowWrapper>
        <TopicDescription title={topic.title} description={topic.description}/>
        <TrainersList
            trainers={topic.trainers}
            author={topic.author}
            topic={topic}
        />
        <Participants topic={topic} likes={topic.members.length}/>
    </RowWrapper>
);

const RowWrapper = styled.div`
    width: 100%;
    padding: 70px 0;
    display: flex;
    justify-content: space-between;
    text-align: left;
    border-bottom: 1px solid #e1e1e1;
`;

export default Topic;
