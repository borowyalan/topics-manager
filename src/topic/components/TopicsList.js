import React from 'react';
import Text from '../../components/Text';
import Topic from './Topic';
import PropTypes from 'prop-types';
import TopicModel from '../../shared/models/TopicModel';
import ActivityIndicator from '../../shared/components/ActivityIndicator';

const TopicsList = ({ topics, pending }) => (
    <React.Fragment>
        <Text type='primary'>Zg≈Çoszone tematy</Text>
        {pending && <ActivityIndicator/>}
        <ul>
            {topics.map((topic, index) => <Topic key={index} topic={topic}/>)}
        </ul>
    </React.Fragment>
);

TopicsList.propTypes = {
    topics: PropTypes.arrayOf(PropTypes.instanceOf(TopicModel)).isRequired,
    pending: PropTypes.bool
};

TopicsList.defaultProps = {
    topics: [],
    pending: false
};

export default TopicsList;
