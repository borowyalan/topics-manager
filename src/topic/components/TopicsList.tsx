import React, { SFC } from 'react';
import styled from 'styled-components';
import Text from '../../components/Text';
import ActivityIndicator from '../../shared/components/ActivityIndicator';
import { Topic as TopicInterface } from '../../shared/models/topic';
import Topic from './Topic';

export interface TopicListProps {
    topics: TopicInterface[],
    pending: boolean,
}

const HeadingText = styled(Text)`
    margin: 20px 0;
`;

const TopicsList: SFC<TopicListProps> = ({ topics = [], pending = false }) => (
    <React.Fragment>
        <HeadingText type="header">
            Zg≈Çoszone tematy
        </HeadingText>
        {pending && <ActivityIndicator/>}
        {topics.map((topic, index) => <Topic key={index} topic={topic}/>)}
    </React.Fragment>
);

export default TopicsList;
