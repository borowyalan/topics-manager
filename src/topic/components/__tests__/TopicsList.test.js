import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import TopicsList from '../TopicsList';
import Topic from '../Topic';
import Topic from '../../../shared/models/topic';
import ActivityIndicator from '../../../shared/components/ActivityIndicator';

configure({ adapter: new Adapter() });

describe('<TopicsList>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TopicsList/>);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have no topics by default', () => {
        expect(wrapper.find(Topic).length).toEqual(0);
    });

    it('should not show pending state by default', () => {
        expect(wrapper.find(ActivityIndicator).length).toEqual(0);
    });

    describe('when topics are provided', () => {
        const topics = [
            TopicModel.fromBackendData({}),
            TopicModel.fromBackendData({})
        ];
        beforeEach(() => {
            wrapper = shallow(<TopicsList topics={topics}/>);
        });

        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should have exactly so much elements as passed to <TopicsList>', () => {
            expect(wrapper.find(Topic).length).toBe(topics.length);
        });
    });

    describe('when topics are loading', () => {
        beforeEach(() => {
            wrapper = shallow(<TopicsList pending={true}/>);
        });

        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

});
