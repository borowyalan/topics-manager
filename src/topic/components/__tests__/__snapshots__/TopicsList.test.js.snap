// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicsList> should render 1`] = `
<React.Fragment>
  <Text
    type="primary"
  >
    Zgłoszone tematy
  </Text>
</React.Fragment>
`;

exports[`<TopicsList> when topics are loading should match snapshot 1`] = `
<React.Fragment>
  <Text
    type="primary"
  >
    Zgłoszone tematy
  </Text>
  <ActivityIndicator />
</React.Fragment>
`;

exports[`<TopicsList> when topics are provided should match snapshot 1`] = `
<React.Fragment>
  <Text
    type="primary"
  >
    Zgłoszone tematy
  </Text>
  <Topic
    key="0"
    topic={TopicModel {}}
  />
  <Topic
    key="1"
    topic={TopicModel {}}
  />
</React.Fragment>
`;
